lib/livebook/utils.ex

```diff
@@ -27,6 +27,50 @@ defmodule Livebook.Utils do
     :crypto.strong_rand_bytes(42) |> Base.url_encode64() |> String.to_atom()
   end
 
+  @doc """
+  Generates a random binary id that includes node information.
+
+  ## Format
+
+  The id is formed from the following binary parts:
+
+    * 16B - hashed node name
+    * 9B - random bytes
+
+  The binary is base32 encoded.
+  """
+  @spec random_node_aware_id() :: id()
+  def random_node_aware_id() do
+    node_part = node_hash(node())
+    random_part = :crypto.strong_rand_bytes(9)
+    binary = <<node_part::binary, random_part::binary>>
+    # 16B + 9B = 25B is suitable for base32 encoding without padding
+    Base.encode32(binary, case: :lower)
+  end
+
+  # Note: the result is always 16 bytes long
+  defp node_hash(node) do
+    content = Atom.to_string(node)
+    :crypto.hash(:md5, content)
```

Comment:
```suggestion
    :erlang.md5(content)
```

Same thing but no crypto dependency.